### DISCUSSION TITLE
Application configuration

### DISCUSSION THREAD

**Discussion by rockerbacon** [rockerbacon/ice-cream-machine-core#17] [Technical]
As made evident in #10, there's a need for configuring a few application-wide values (HTTP server host and port, in that instance).

**Comment by rockerbacon** [comment]
I'd suggest a simple 2 level configuration:
1. Configurations are first read from a configuration file
2. If a value isn't specified in the configuration file, a sensible default defined in the code itself is used

The configuration file should be read from a default configuration location, eg.: `~/.config/icm/settings.json`. Having an optional argument to specify a different configuration location would be a good idea.
For the file format, I'd suggest JSON with all values at the top level, eg.: `{"port":8080,"host":"localhost"}`. JSON because it keeps uniformity with everything else in the application and everything at the top level because I don't anticipate too many parameters are necessary (specially when we're talking about having sensible defaults for everything).

**Comment by rockerbacon** [comment]
I thought a little about environment variables but couldn't find a practical use-case for them, considering my 2 level solution above.

**Reply by nlaranjeira01** [reply]
I could only think of environment variables in the case of overriding the configuration file location for testing different configurations, but it doesn't look very useful when I realize that we won't have that many configuration options anyway.

**Comment by nlaranjeira01** [comment]
No problem with that. Agreed.👍
